src/systemctl/sysv-compat.c

```diff
@@ -73,30 +73,40 @@ int parse_shutdown_time_spec(const char *t, usec_t *ret) {
 
                 errno = 0;
                 hour = strtol(t, &e, 10);
-                if (errno > 0 || *e != ':' || hour < 0 || hour > 23)
+                if (errno > 0 || *e != ':' || hour < 0 || hour > 99999)
```

Comment:
This seems like an arbitrary restriction. Why not remove it completely?

Reply:
There is a non-arbitrary upper limit where the dates will fail to resolve, and return 
"Shutdown scheduled for --- XXXX-XX-XX XX:XX:XX, use 'shutdown -c' to cancel."
You can test this with `shutdown -k +9999999999`.
It could be higher than 99999, but I think the point where it fails is at a particular date, not a particular distance away from the current date, so the absolute limit will keep moving downward as time progresses. I picked 99999 because it's a little over 10 years and seemed like long enough that it would never be used, while being far enough from the failing point that it won't break by the time all of us are dead.